model/pod/pod.go

```diff
@@ -87,3 +121,44 @@ func (p *Pod) Remove() error {
 		},
 	)
 }
+
+// UpdateStatus updates the pod status.
+func (p *Pod) UpdateStatus(s Status) error {
+	byIDAndStatus := ByID(p.ID)
+	byIDAndStatus[StatusKey] = p.Status
```

Comment:
do we only want to update the status when we have the correct status in the db?

Reply:
What do you mean? The point of this is to verify that the in-memory model hasn't gone stale and to guarantee that the state transition is atomic.